### ISSUE TITLE
dynamic user: mislabeled directory

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#7741]
### Submission type

 - Bug report

### systemd version the issue has been seen with

236

### Used distribution

Fedora Rawhide

### In case of bug report: Expected behaviour you didn't see

Proper label on /var/lib//private/systemd/timesync

### In case of bug report: Unexpected behaviour you saw

Mislabeled /var/lib/private/systemd/timesync

### In case of bug report: Steps to reproduce the problem
```
matchpathcon /var/lib/systemd/timesync
matchpathcon /var/lib/private/systemd/timesync
ls -dZ /var/lib/private/systemd/timesync
```
systemd does not create the /var/lib/private/systemd/timesync directory with the label that is returned by `matchpathcon /var/lib/private/systemd/timesync`

the directory is instead labeled with the context of the parent directory /var/lib/private/systemd

**Comment by ghost** [comment]
```
# ls -alZ /var/lib/systemd/timesync/
total 8
drwxr-xr-x. 2 systemd-timesync systemd-timesync sys.id:sys.role:systemd.generic_var_lib.var_lib_file:s0 4096 Dec 25 10:02 .
drwxr-xr-x. 3 root             root             sys.id:sys.role:systemd.generic_var_lib.var_lib_file:s0 4096 Dec 25 10:02 ..
-rw-r--r--. 1 systemd-timesync systemd-timesync sys.id:sys.role:timesyncd.var_lib.var_lib_file:s0          0 Dec 25 10:35 clock
```

```
# matchpathcon /var/lib/systemd/timesync/
/var/lib/systemd/timesync       sys.id:sys.role:timesyncd.var_lib.var_lib_file:s0
```

```
# matchpathcon /var/lib/private/systemd/timesync/
/var/lib/private/systemd/timesync       sys.id:sys.role:timesyncd.var_lib.var_lib_file:s0
```

```
# restorecon -RvF /var/lib/private/systemd
Relabeled /var/lib/private/systemd/timesync from sys.id:sys.role:systemd.generic_var_lib.var_lib_file:s0 to sys.id:sys.role:timesyncd.var_lib.var_lib_file:s0
```

**Comment by yuwata** [comment]
I think this is already fixed by #7663.
Duplicate of #7659.

**Comment by ghost** [comment]
Should probably be tagged "bug" then, and the patch be pushed to rawhide. I will close this
